py-polars/pyproject.toml

```diff
@@ -75,6 +75,9 @@ plot = ["hvplot >= 0.9.1", "polars[pandas]"]
 style = ["great-tables >= 0.8.0"]
 timezone = ["backports.zoneinfo; python_version < '3.9'", "tzdata; platform_system == 'Windows'"]
 
+# GPU Engine
+gpu = ["cudf-polars-cu12"]
```

Comment:
TODO: `gpu` should only be a valid option on WSL and Linux (not macos or windows).

Reply:
I think it's just better to let pip try to find a Windows wheel and fail.
(as long as we don't end up in a degenerate case where pip backtracks forever)

Not sure if there is a way to make extras platform specific, but I think having pip fail is better than having polars[gpu] install nothing.

Reply:
I think it'd be a good idea to clarify this in the docstring.

Reply:
You can do this to add a platform-specific dependency, but it's probably not what you want. You probably want a hard error on Windows if the user tries to install with `gpu`, whereas this suggestion would silently install without GPU support.
```
gpu = ["cudf-polars-cu12; platform_system == 'Linux'"]
```

Reply:
You can't make extras platform-specific AFAIK. You could make the packages listed in the extra platform-specific using platform selectors (like the one for `tzdata` above), but I don't think that's desirable because then you'd silently have `pip install polars[gpu]` be a no-op and then get user confusion when nothing is running on the GPU.

Reply:
Yep, I think that was the intent of my original comment.

Sounds like we're all in agreement here!

Reply:
Resolving as nothing to do.